src/states/client.rs

```diff
@@ -251,9 +278,12 @@ impl Client {
                          priority: u8)
                          -> Result<(), RoutingError> {
         self.stats.count_user_message(&user_msg);
-        for part in user_msg.to_parts(priority)? {
+        let (hash, parts) = user_msg.to_parts(priority)?;
```

Comment:
The hash returned is only used here. Wouldn't it be cleaner to implement a `get_hash` method on `UserMessage` and call it here, and go back to only returning the parts from `to_parts`? The parts contain the hash, too, so maybe reading it from any of the parts would also be an option?